docs/design/2026-07-14-trim-minmax-for-date-types.md

```
@@ -0,0 +1,982 @@
+```cpp
+const UInt8 pack_mark = trim_index.minmax->packMark(pack_id);
+const bool has_null = (pack_mark & 0x01) != 0;
+const bool has_trimmed_low = (pack_mark & 0x02) != 0;
+const bool has_trimmed_high = (pack_mark & 0x04) != 0;
+```
+
+Based on the predicate class, it determines whether a trimmed value that must match or must not match exists:
+
+| Predicate type | `trimmed_match_exists` | `trimmed_nonmatch_exists` |
+| --- | --- | --- |
+| Equality / IN / bounded range with `Q ⊆ E` | false | `has_trimmed_low || has_trimmed_high` |
```

markdownlint-cli2 (0.23.2)

[warning] Table column count
Expected: 3; Actual: 5; Too many cells, extra data will be missing

(MD056, table-column-count)